doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,436 @@
+
+Adding an upgradable bootloader
+*******************************
+
+MCUboot is the only upgradable bootloader currently available with the |NCS|.
+The following section describes how to add it to your secure bootloader chain.
+
+.. _ug_bootloader_adding_upgradable_mcuboot:
+
+MCUboot
+=======
+
+To use MCUboot as an upgradable bootloader, the application must include the |NSIB| as the immutable bootloader.
+To do so, add :option:`CONFIG_BOOTLOADER_MCUBOOT` with either the build command or in the application's :file:`prj.conf` file:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y
+
+|how_to_configure|
+
+MCUboot can use the cryptographic functionality exposed by the immutable bootloader, reducing the flash memory usage for MCUboot to less than 16 kB.
+To enable this configuration, apply the :file:`overlay-minimal-external-crypto.conf` Kconfig fragment file for the MCUboot image:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf5840 zephyr/samples/hello_world -- \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -Dmcuboot_OVERLAY_CONFIG=overlay-minimal-external-crypto.conf
+
+See :ref:`ug_bootloader_adding_fragments` for more information about using Kconfig fragments with bootloaders.
+
+Adding an upgradable bootloader to an application changes its flash memory partition layout to accommodate first and second image slots for both immutable and upgradable bootloaders.
+See :ref:`ug_bootloader_flash_mcuboot` for more information about how this bootloader organizes its flash memory layout.
+
+Signature keys
+--------------
+
+The process to use specific signature keys with MCUboot used as the upgradable bootloader is the same as when it is used :ref:`as the immutable one <ug_bootloader_adding_immutable_mcuboot_keys>`.
+
+.. note::
+
+   Since each bootloader is built with its own signature key, using a different private key with an upgradable bootloader will not cause problems with the secure boot chain.
+   You can also use the same private key for both the immutable and upgradable bootloaders, as long as the key type is supported by both of them.
+
+.. _ug_bootloader_adding_presigned_variants:
+
+Pre-signed variants
+===================
+
+Enable the :option:`CONFIG_BUILD_S1_VARIANT` option when building the upgradable bootloader to automatically generate :ref:`pre-signed variants <upgradable_bootloader_presigned_variants>` of the image for both slots:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -DCONFIG_BUILD_S1_VARIANT=y
+
+This is a necessary step for creating application update images for use with :ref:`ug_fw_update`.
+
+.. _ug_bootloader_adding_testing:
+
+Testing the boot chain
+**********************
+
+To test the bootloader(s) and the secure boot chain, build and program them with any sample as described above.
```

Comment:
This formatting "the bootloader(s)" is not allowed. Suggestion: `either of the bootloaders`.

Also, I would move this and the following sections (that is, Testing, Bootloader configs, and External partitions) to separate pages. As we're going for container-based structure on the top page of the bootloader section, we should be consistent. Testing is a separate procedure from Adding and the rest of information in the other two sections.